src/lib/JIT.hs

```diff
@@ -777,6 +783,7 @@ callableOperand ty name = Right $ L.ConstantOperand $ C.GlobalReference ty name
 
 asLLVMName :: Name -> L.Name
 asLLVMName name@(Name TopFunctionName _ _) = fromString $ pprint name
+asLLVMName (GlobalName tag) = fromString $ "__" ++ pprint tag
```

Comment:
When do we use that?

Reply:
Non-inlined functions live in the `GlobalName` namespace. The underscores are there to distinguish them from `TopFunctionName` names. Ideally we'd have a proper invertible `Name -> L.Name` mapping. I'll add a comment.

Reply:
Why can't we use `TopFunctionName`s for those as well? This is why this namespace exists after all